src/python/WMCore/MicroService/MSOutput/MSOutput.py

```diff
@@ -118,20 +116,32 @@ def __init__(self, msConfig, mode, logger=None):
         self.campaigns = {}
         self.psn2pnnMap = {}
 
+        self.msConfig.setdefault("mongoDBUrl", 'mongodb://localhost')
+        self.msConfig.setdefault("mongoDBPort", 8230)
+        self.msConfig.setdefault("mongoDB", 'msOutputDBDefault')
```

Comment:
> I am inclined to think that the first 3 parameters should not have any default. It MUST be provided in the configuration, and if a given person/setup fails to do so, things should simply break or not even start.

This would lead to an immediate disaster. Because MongoDB never fails a connection and is absolutely silent even for a missing database or collection (it just recreates the database/collection on connection attempt if it is missing). Those defaults here are provided for this script to protect us from human errors such that if one makes a mistake and tries to connect without providing those, then the script would just safely default to a blank set of parameters that would not lead to interference with the any production or private database instance. And since we are connecting several services (including testbed and private intance from dev K8 clusters) to a single MongoDB cluster with a single replicaset and destinguishing between service instances only by database names, I'd absolutely sure if we fail to provide a safe set of defaults here to point to a null database and collection someone early or late would make a mistake and would wipe out someone else's database/collection or the testbed ones.  Even more, currently we are in that same situation even for the production `MongoDBAAS` cluster because the testbed instance is not yet ready.

Reply:
> With that said, I would suggest as well to update all the 3 branches in services_config repository to explicitly define the mongo database name (mongoDB attr), given that we only have the url and the port number at this moment

I do not get what you mean here. It is already done and the relevant merge requests to `services_config` repository are already in the PR's description.

Reply:
>This would lead to an immediate disaster. Because MongoDB never fails a connection and is absolutely silent even for a missing database or collection (it just recreates the database/collection on connection attempt if it is missing).

This is precisely my point! I prefer a disaster - which in this context would be that every mongodb document would fail to get written - than silently writing to a different database and/or instance of MongoDB.

In other words, we need to make sure that the service IS using the MongoDB configuration as provided in the configuration file. If it's missing from the configuration file, then the service should fail to start up.

In addition to `mongoDBUrl`, `mongoDBPort` and `mongoDB`, it looks like we should also pass the `collections` parameter (likely better to rename it to `mongoCollections` or similar.

In case any of those are missing in the configuration file, MongoDB will fail to initialize - with a KeyError exception - because of lines like:
```
  'database': self.msConfig['mongoDB']
```

making it clear that there is a configuration problem with the service.